processor/processor.go

```diff
@@ -1259,26 +1279,67 @@ func (proc *HandleT) processJobsForDest(jobList []*jobsdb.JobT, parsedEventList
 		}
 	}
 
+	if len(statusList) != len(jobList) {
+		panic(fmt.Errorf("len(statusList):%d != len(jobList):%d", len(statusList), len(jobList)))
+	}
+
+	return transformationMessage{
+		groupedEvents,
+		trackingPlanEnabledMap,
+		eventsByMessageID,
+		uniqueMessageIdsBySrcDestKey,
+		reportMetrics,
+		statusList,
+		procErrorJobs,
+		sourceDupStats,
+		uniqueMessageIds,
+
+		totalEvents,
+		start,
+	}
+}
+
+type transformationMessage struct {
+	groupedEvents map[string][]transformer.TransformerEventT
+
+	trackingPlanEnabledMap       map[SourceIDT]bool
+	eventsByMessageID            map[string]types.SingularEventWithReceivedAt
+	uniqueMessageIdsBySrcDestKey map[string]map[string]struct{}
+	reportMetrics                []*types.PUReportedMetric
+	statusList                   []*jobsdb.JobStatusT
+	procErrorJobs                []*jobsdb.JobT
+	sourceDupStats               map[string]int
+	uniqueMessageIds             map[string]struct{}
+
+	totalEvents int
+	start       time.Time
+}
+
+func (proc *HandleT) transformations(in transformationMessage) storeMessage {
 	//Now do the actual transformation. We call it in batches, once
 	//for each destination ID
 
+	var procErrorJobsByDestID = make(map[string][]*jobsdb.JobT)
+	var batchDestJobs []*jobsdb.JobT
+	var destJobs []*jobsdb.JobT
+
 	destProcStart := time.Now()
 	proc.logger.Debug("[Processor: processJobsForDest] calling transformations")
```

Comment:
What is this debug log for? Do we need to update the log to reflect new function name? Do we need the log at all?